nvdaHelper/microsoft-ui-uiautomation/sconscript

```diff
@@ -0,0 +1,106 @@
+###
+#This file is a part of the NVDA project.
+#URL: http://www.nvaccess.org/
+#Copyright 2021 NV Access Limited
+#This program is free software: you can redistribute it and/or modify
+#it under the terms of the GNU General Public License version 2.0, as published by
+#the Free Software Foundation.
+#This program is distributed in the hope that it will be useful,
+#but WITHOUT ANY WARRANTY; without even the implied warranty of
+#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#This license can be found at:
+#http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+###
+
+import os
+import glob
+
+Import(
+	'env',
+	'sourceDir',
+	'sourceLibDir',
+)
+
+MSUIA_sourceDir = Dir('#include/microsoft-ui-uiautomation/src/uiAutomation')
+MSUIA_lib_outDir = Dir('lib')
+MSUIA_include_outDir = Dir('include')
+MSUIA_solutionFile = MSUIA_sourceDir.File('UIAutomation.sln')
+
+MSUIA_wrapper_libs = env.Command(
+	target = [
+		MSUIA_lib_outDir.File('Microsoft.UI.UIAutomation.dll'),
+		MSUIA_lib_outDir.File('Microsoft.UI.UIAutomation.exp'),
+		MSUIA_lib_outDir.File('Microsoft.UI.UIAutomation.pdb'),
+		MSUIA_lib_outDir.File('Microsoft.UI.UIAutomation.lib'),
+		MSUIA_lib_outDir.File('winmd/Microsoft.UI.UIAutomation.winmd'),
+		MSUIA_sourceDir.File('microsoft.ui.uiautomation/Generated Files/winrt/microsoft.ui.uiautomation.h'),
+		MSUIA_sourceDir.File('microsoft.ui.uiautomation/Generated Files/winrt/impl/microsoft.ui.uiautomation.0.h'),
+		MSUIA_sourceDir.File('microsoft.ui.uiautomation/Generated Files/winrt/impl/microsoft.ui.uiautomation.1.h'),
+		MSUIA_sourceDir.File('microsoft.ui.uiautomation/Generated Files/winrt/impl/microsoft.ui.uiautomation.2.h'),
+	],
+	source = [
+		MSUIA_sourceDir.File('microsoft.ui.uiautomation/Microsoft.UI.UIAutomation.vcxproj'),
+		MSUIA_sourceDir.File('microsoft.ui.uiautomation/Microsoft.UI.UIAutomation.idl'),
+		glob.glob(os.path.join(MSUIA_sourceDir.abspath, 'microsoft.ui.uiautomation', '*.cpp')),
+		glob.glob(os.path.join(MSUIA_sourceDir.abspath, 'microsoft.ui.uiautomation', '*.h')),

```

Comment:
As these globs are only to find source files for the sake of checking if they are out of date (and a build is required), including too many is worse than missing some. Note that this list is never passed to a compiler for actual building. I would much prefer to keep the glob lines, rather than list out all 30 or more source files, which may very well get added to and we wouldn't know... it just wouldn't rebuild if those new ones changed.

Reply:
Ok that is fine. The `msbuild` in the action looks like these source files are used for building though?

Reply:
the msbuild command references `$SOURCE`. This variable only ever contains the first source file. You'd need to use `$SOURCES` to have all of them appear in the command.